test/evp_test.c

```diff
@@ -2859,7 +2859,9 @@ static int kdf_test_ctrl(EVP_TEST *t, EVP_KDF_CTX *kctx,
     if (!TEST_ptr(name = OPENSSL_strdup(value)))
         return 0;
     p = strchr(name, ':');
-    if (p != NULL)
+    if (p == NULL)
+        p = "";
```

Comment:
Hmm, okay, but this trades a bunch of `NULL` tests for a situation where `*p = 'x'` will now segfault.

Maybe a `const char *value` that gets set to something that is always usable?

Since this is tests, I'll approve this PR anyway, and leave it up to you if you want to change it

Reply:
I think we're ok here, given that all call sites that pass p in this path do so as const char * already, so it shouldn't ever be modified

Reply:
Depends what you mean by "this path", given the (new) line 2865 has `*p++ = '\0'`!